src/map.js

```diff
@@ -0,0 +1,21 @@
+const map = ( array, operation ) => {
+  const fn = operation || ( () => {} )
+  let results =[]
+  if(typeof operation === 'function'){
```

Comment:
For clarity, I would separate these two different map approaches into their own functions, and then `map` could look something like:
```es6
return {
  function: mapWithFunction,
  string: mapWithString
}[ typeof operation ]()
```